### PR TITLE
feat: use optimistic oracle v2 and simplify logic

### PR DESCRIPTION
**Pull request by md0x**
Signed-off-by: Pablo Maldonado <pablo@umaproject.org>

**Motivation**

Address remaining comments from : https://github.com/UMAprotocol/dev-quickstart/pull/1#discussion_r895928121

**Summary**
- Use new OptimisticOracleV2
- Remove unnecessary LSP logic

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [ ]  Ran end-to-end test, running the code as in production
- [x]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [x]  All existing tests pass
- [ ]  Untested

### CODE REVIEW

File: contracts/EventBasedPredictionMarket.sol
Comment (chrismaree) [inline contracts/EventBasedPredictionMarket.sol]: we will be able to replace this shortly :)

Comment (chrismaree) [reply contracts/EventBasedPredictionMarket.sol]: actually, I'm confused. where is this `OptimisticOracleV2Interface.sol` file? it might make sense for us to wait until we've done the next release before merging this PR

File: contracts/EventBasedPredictionMarket.sol
Comment (chrismaree) [inline contracts/EventBasedPredictionMarket.sol]: this is confusing. please refactor to avoid double nested ternary operator. also your comment has a typo in it (wether)

**Review by chrismaree — COMMENTED** [review]
this is awesome! I Like the simplifcation you did here. I think we should hold off on merging this until we've done another protocol release then you can import the "correct" OOV2 interface from the `core` export

File: contracts/EventBasedPredictionMarket.sol
Comment (md0x) [reply contracts/EventBasedPredictionMarket.sol]: I have a local copy of the contracts that I have not put into this PR so as not to introduce unnecessary noise. As soon as there is a release I will be able to change that line.

File: contracts/EventBasedPredictionMarket.sol
Comment (smb2796) [inline contracts/EventBasedPredictionMarket.sol:81]: We should also verify that the collateralToken is a registered collateral type, since it is used as part of price request parameters.

**Comment by smb2796** [comment]
Agreed with Chris! Much simpler and understandable. Good stuff @md0x

File: contracts/EventBasedPredictionMarket.sol
Comment (chrismaree) [inline contracts/EventBasedPredictionMarket.sol:11]: <3

**Review by chrismaree — APPROVED** [review]

**Pull request merged**
